sound/soc/sof/sof-audio.c

```diff
@@ -348,15 +348,16 @@ static int sof_free_widgets_in_path(struct snd_sof_dev *sdev, struct snd_soc_dap
 	int err;
 	int ret = 0;
 
+	if (WIDGET_IS_AIF_OR_DAI(widget->id)) {
```

Comment:
The existing code test `widget->dobj.private` before calling `sof_widget_free(sdev, widget->dobj.private);`. Should we do the same test when we move `sof_widget_free(sdev, widget->dobj.private);` out of the loop?

Reply:
good point. added the check now